### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by DavidNew-NOAA**
This PR adds a block to the 3dvar and local ensemble DA YAMLs to allow those applications to act as application tests, as long as "test_file" is defined in the JCB prototype YAML. This will be turned on in jjob testing in a future GDASApp PR.

### CODE REVIEW

File: 3dvar.yaml.j2 [hunk 1]
```diff
@@ -98,3 +98,12 @@ output:
...
 {% endif %}
+
+# Optionally test the application
+{% if test_file is defined %}
+test:
+{% filter indent(width=2) %}
+{% set test_file = test_file|default('test', true) ~ '.yaml.j2' %}
+{% include test_file %}
+{% endfilter %}
+{% endif %}
```
Comment (CoryMartin-NOAA) [inline 3dvar.yaml.j2]: emacs strikes again with no EOL

Comment (DavidNew-NOAA) [reply 3dvar.yaml.j2]: 🤬

**Review by CoryMartin-NOAA — APPROVED** [review]

**Review by danholdaway — APPROVED** [review]

**Review by danholdaway — CHANGES_REQUESTED** [review]
Looking at the PRs I realize that we could do this without the step of changing jcb-gdas. Instead of using a test file you can replace what you have in job-algorithms with:

``` yaml
# Optionally test the application
 {% if test_reference_filename is defined %}
 test:
   reference filename: {{test_reference_filename}}
{% if test_output_filename is defined %}
   test output filename: {{test_output_filename}}
{% endif %}
{% if log_output_filename is defined %}
   log output filename: {{log_output_filename}}
{% endif %}
{% if mpi_pattern is defined %}
   mpi pattern: {{mpi_pattern}}
{% endif %}
   float relative tolerance: {{test_float_relative_tolerance | default(1.0e-6, true)}}
   float absolute tolerance: {{test_float_absolute_tolerance | default(0.0, true) }}
   integer tolerance: {{test_integer_tolerance | default(0, true) }}
 {% endif %}
 ```

Note that I've accounted for the complete logic [here](https://github.com/JCSDA/oops/blob/7e93aa5e1f3b7f41740ae86841ac714b3ec71024/src/oops/util/TestReference.cc#L102). 

Then you only need to change the YAMLs in the GDASapp test directories to include the reference and tolerances. I.e. you need test_reference_filename et al [here](https://github.com/NOAA-EMC/GDASApp/blob/develop/test/atm/global-workflow/jcb-prototype_3dvar.yaml.j2).

This actually makes more sense because there isn't really a sensible place to add that test file. You correctly put it in the atmosphere directory of jcb-gdas but we'd put the exact same file in the marine, aero etc directories. This way we avoid any changes to jcb-gdas.

Note that we could make it so you don't need to add the above YAML to every file in job-algorithms and instead have a global option to add it to any algorithm but perhaps for later.

**Comment by DavidNew-NOAA** [comment]
@danholdaway Done. Makes sense to keep things more generic, thanks

**Review by danholdaway — APPROVED** [review]

**Pull request merged**